### PR TITLE
Friend icons show on load + loading screen

### PR DESCRIPTION
**Pull request by carolinecreamer**
In the feed tab, users that the current user follows show on the feed tab. The "friend" state is updated automatically. Additionally, a "loading" page appears when the application is loading.

Changes:
- "Friend" state is updated to reflect if the user follows another user in the DB
- On page load, friend icons are updated to reflect "friend" state
- Loading page shows when the page is loading

Next Steps:
- Friends appear on map
- Nearest friend algorithm

Callouts For Review:
- N/A 

Resources Used:
- [Bootstrap Spinner Docs](https://react-bootstrap.github.io/components/spinners/)
- [Center items on screen](https://www.w3.org/Style/Examples/007/center.en.html): Used to center loading spinner on screen
- [Place friend icon in top right corner](https://www.quora.com/How-can-you-position-an-element-in-the-top-right-corner-of-a-card)

Before:
Screen Shot 2022-07-26 at 10 44 13 AM

After:
Screen Shot 2022-07-26 at 10 43 56 AM
Screen Shot 2022-07-26 at 10 13 44 AM

### CODE REVIEW

File: ui/src/components/UserProfile/UserProfile.css [hunk 1]
```diff
@@ -20,3 +20,8 @@
 .favorites {
```
Comment (iwen-fb) [inline ui/src/components/UserProfile/UserProfile.css]: fyi: when you are applying the same styling to mulitple classes you can do something like 
```
.favorites, .following { 
     font-weight: bold;
}

File: api/userRoutes/users.js [hunk 2]
```diff
@@ -16,11 +16,11 @@ router.get("/get-users", async (req, res, next) => {
 
 router.get("/get-following", async (req, res, next) => {
     try {
-        const user = await User.getUser();
-        const query = new Parse.Query("Follow");
-        query.equalTo("from", user);
-        const users = await query.find();
-        res.status(200).json({ users })
+        const currentUser = await User.getUser();
+        let followingPointers = currentUser.get("friends");
```
Comment (iwen-fb) [inline api/userRoutes/users.js:20]: should these be `const`?

**Review by iwen-fb — APPROVED** [review]
a couple nits- but the functionality looks great!
image

**Comment by iwen-fb** [comment]
I think there are likely some merge conflicts/files included in this change that weren't supposed to be - i noticed it's a branch of main - to resolve this you probably want to rebase this branch on the one in this pr: https://github.com/carolinecreamer/CapstoneProject/pull/35 ?

File: ui/src/components/UserProfile/UserProfile.css (hunk 1 above)
Comment (carolinecreamer) [reply ui/src/components/UserProfile/UserProfile.css]: Good to know! Thank you!

File: api/userRoutes/users.js (hunk 2 above)
Comment (carolinecreamer) [reply api/userRoutes/users.js:20]: Just changed! and just rebased onto pr #35 to make sure there are no merge conflicts

**Pull request merged**